plugins/catalog-backend/src/ingestion/LocationReaders.ts

```diff
@@ -162,25 +166,65 @@ export class LocationReaders implements LocationReader {
   private async handleEntity(
     item: CatalogProcessorEntityResult,
     emit: CatalogProcessorEmit,
-  ): Promise<Entity> {
+  ): Promise<Entity | undefined> {
     const { processors, logger } = this.options;
 
     let current = item.entity;
 
+    // Construct the name carefully, this happens before validation below
+    // so we do not want to crash here due to missing metadata or so
+    const kind = current.kind || '';
+    const namespace = !current.metadata
+      ? ''
+      : current.metadata.namespace ?? ENTITY_DEFAULT_NAMESPACE;
+    const name = !current.metadata ? '' : current.metadata.name;
+
+    for (const processor of processors) {
+      if (processor.preProcessEntity) {
+        try {
+          current = await processor.preProcessEntity(
+            current,
+            item.location,
+            emit,
+          );
+        } catch (e) {
+          const message = `Processor ${processor.constructor.name} threw an error while preprocessing entity ${kind}:${namespace}/${name} at ${item.location.type} ${item.location.target}, ${e}`;
+          emit(result.generalError(item.location, message));
+          logger.warn(message);
+          return undefined;
```

Comment:
At this point i chose to abort processing if any processor throws. It could be argued that processing could continue  - but not knowing the nature of the processors, it's hard to know if that's safe or just makes things worse